rancher2/schema_cluster_v2_rke_config.go

```diff
@@ -112,6 +112,15 @@ func clusterV2RKEConfigFieldsV0() map[string]*schema.Schema {
 				Schema: clusterV2RKEConfigSystemConfigFieldsV0(),
 			},
 		},
+		"machine_selector_files": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Computed:    true,
+			Description: "Cluster V2 machine selector files",
+			Elem: &schema.Resource{
+				Schema: clusterV2RKEConfigMachineSelectorFilesFields(),
+			},
+		},
```

Comment:
As we are adding new fields do we need to add them for both schema and v0 schema, ain't we only supposed to add breaking changes and its migration logic?

Honestly Idk but it doesn't seems right. 

@a-blender

Reply:
This is a grey area in this repo.  I agree that, technically, we should not modify the old v0 schema, but in practice, I have seen multiple PRs that make changes to it. ([example1](https://github.com/rancher/terraform-provider-rancher2/pull/1239/files#diff-ff7f9b8a064e715376221a90cbff1c99f633ccb9c5c075dd4398b3bdd4c05c3dR69) [example2](https://github.com/rancher/terraform-provider-rancher2/pull/1119/files#diff-b3e412e91a344e8c58223804e3b9f48b756a385afbbf287c7bd0dfaf85e4da44R405-R410) ).

This makes me think that it is expected, or our practice, to add new **optional** fields to both the v0 and v1 schema. 

WDYT?

Reply:
I believe if we modify a field, we only change the type in the new v1 (or vx) schema and add migration logic so the TF provider knows what to do on upgrade of the provider. Correct, new fields should be added to all schema versions.